src_common/uint128.c

```diff
@@ -0,0 +1,292 @@
+/*******************************************************************************
```

Comment:
Hard file to review, I did my best but ultimately reading math formulas without knowing the context is not very useful
Can we implement unit tests for this sort of files ? To test edge cases very quickly

Reply:
Out of the scope of this PR, this code was only split and isn't new.